File: problem001.py
""" problem001.py:
        Find the sum of all the multiples of 3 or 5 below 1000.
"""
from __future__ import division  # lol python2


def arith(lower, upper):
    while upper%lower:
        upper -= 1
    # print("({} + {})/2 * {}".format(lower, upper, upper/lower))
    return (lower + upper)/2 * (upper/lower)


def main(max_):
    return arith(3, max_) + arith(5, max_) - arith(15, max_)


if __name__ == "__main__":
    print(main(999))

File: javascript/problem001.js
// problem001.js
//  Find the sum of all multiples of 3 or 5 below 1000.

function algebraic(upper, factor) {
    upper = Math.floor((upper - 1) / factor);
    sum = (1+upper) * (upper/2)
    return Math.floor(factor * sum)
}

function main(maximum) {
    return algebraic(maximum, 3) + algebraic(maximum, 5) - algebraic(maximum, 15)
}

console.log(main(1000))
